### PR TITLE
Gradle build integration

### PR DESCRIPTION
**Pull request by subarnob**
Added a sample java project `jar-infer:test-java-jarinfer` to demo jarinfer integration with gradle.

We add a dependency to `"com.uber.nullaway:jar-infer-cli:0.6.0"` and the build script basically scans the classpath and invokes jarinfer for each jar/aar library and adds the model jar to the classpath.

### CODE REVIEW

**Comment by lazaroclapp** [comment]
Hey @subarnob , this is expected to be both an example and a failing integration test, correct? Since the point is that the apache methods have their nullability inferred by jarinfer and thus they should be reporting errors (which are currently being reported above as Travis CI errors).

@msridhar You said we already had some examples of this of integration test running?

**Comment by subarnob** [comment]
@lazaroclapp Yes. The `build.gradle` was meant to be the example, and I totally overlooked the failing CI errors. I'll re-write them as unit tests?

File: jar-infer/test-java-jarinfer/build.gradle [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  if (!name.toLowerCase().contains("test")) {
+    options.compilerArgs += ["-Xep:NullAway:ERROR", "-XepOpt:NullAway:AnnotatedPackages=com.uber", "-XepOpt:NullAway:JarInferUseReturnAnnotations=true", "-XepOpt:NullAway:JarInferRegexStripModelJar=.*/([^/]+)\\.astubx\\.[ja]ar\$" ]
+  }
+}
+
+def jiPath = "com.uber.nullaway/jar-infer-cli"
+def libPath = ""
+def modelPath = ""
+compileJava.dependsOn configurations.compileClasspath
+compileJava.doFirst {
```
Comment (subarnob) [inline jar-infer/test-java-jarinfer/build.gradle]: @lazaroclapp Do you have suggestions on this part? This code would need to be appended to the Gradle build script for projects. Is there an elegant way of doing this?

**Comment by msridhar** [comment]
No longer relevant

**Pull request closed without merging**